### ISSUE TITLE
How about always compress coredump before uploading?

### ISSUE THREAD

**Issue by kongove** [scylladb/scylla-cluster-tests#270]
Right now if the coredump is less than 5G, we will upload it directly without compressing.

How about always compress the coredump before uploading for saving the space? It only effect the core file that is less than 5G.

In one [longevity job 45](http://jenkins.cloudius-systems.com/job/scylla-1.7-longevity-gce/45/label=gce/), it generated a 92G coredump, it was compressed to 2.6G and uploaded without splitting.

```
diff --git a/sdcm/cluster.py b/sdcm/cluster.py
index 536c6d5..a2377d2 100644
--- a/sdcm/cluster.py
+++ b/sdcm/cluster.py
@@ -689,7 +689,7 @@ WantedBy=multi-user.target
             self.log.error('Failed getting coredump file size: %s', ex)
         return None
 
-    def _split_coredump(self, coredump):
+    def _try_split_coredump(self, coredump):
         core_files = []
         try:
             self.remoter.run('sudo yum install -y pigz')
@@ -723,10 +723,7 @@ WantedBy=multi-user.target
                 coredump = line.split()[-1]
                 self.log.debug('Found coredump file: {}'.format(coredump))
                 file_size = self._get_coredump_size(coredump)
-                if file_size and file_size > COREDUMP_MAX_SIZE:
-                    coredump_files = self._split_coredump(coredump)
-                else:
-                    coredump_files = [coredump]
+                coredump_files = self._try_split_coredump(coredump)
                 for f in coredump_files:
                     self._upload_coredump(f)
                 if len(coredump_files) > 1 or coredump_files[0].endswith('.gz'):
```

**Comment by larisau** [comment]
yes, we could do it. We need to remove the file size check in two places - before compression and before splitting.

**Comment by roydahan** [comment]
I'm fine with that.
@larisau, please add it to your queue.

**Comment by lmr** [comment]
+1. More than 40x compression should compensate for the compression time.

**Comment by kongove** [comment]
Solved by PR #271
